.buildkite/scripts/steps/code_coverage/ftr_configs.sh

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common/util.sh
+
+.buildkite/scripts/bootstrap.sh
+.buildkite/scripts/build_kibana_plugins.sh
+
+is_test_execution_step
+
+export JOB_NUM=$BUILDKITE_PARALLEL_JOB
+export JOB=ftr-configs-${JOB_NUM}
+export CODE_COVERAGE=1
+
+FAILED_CONFIGS_KEY="${BUILDKITE_STEP_ID}${BUILDKITE_PARALLEL_JOB:-0}"
+
+# a FTR failure will result in the script returning an exit code of 10
+exitCode=0
+
+configs="${FTR_CONFIG:-}"
+
+if [[ "$configs" == "" ]]; then
+  echo "--- downloading ftr test run order"
+  buildkite-agent artifact download ftr_run_order.json .
+  configs=$(jq -r '.groups[env.JOB_NUM | tonumber].names | .[]' ftr_run_order.json)
+fi
+
+failedConfigs=""
+results=()
+
+while read -r config; do
+  if [[ ! "$config" ]]; then
+    continue
+  fi
+
+  echo "--- $ node scripts/functional_tests --config $config --exclude-tag ''skipCoverage''"
+  start=$(date +%s)
+
+  # prevent non-zero exit code from breaking the loop
+  set +e
+  NODE_OPTIONS=--max_old_space_size=14336 \
+    ./node_modules/.bin/nyc \
+    --nycrc-path ./src/dev/code_coverage/nyc_config/nyc.server.config.js \
+    node scripts/functional_tests \
+    --config="$config" \
+    --exclude-tag "skipCoverage"
+  lastCode=$?
+  set -e
+
+  dasherize() {
+    withoutExtension=${1%.*}
+    dasherized=$(echo "$withoutExtension" | tr '\/' '\-')
+  }
+  dasherize $config
+
+  serverAndClientSummary="target/kibana-coverage/functional/xpack-${dasherized}-server-coverage.json"
+  functionalSummary="target/kibana-coverage/functional/xpack-${dasherized}-coverage.json"
+
+  # Server side and client side (server and public dirs)
+  if [[ -d "$KIBANA_DIR/target/kibana-coverage/server" ]]; then
+    echo "--- Server and Client side code coverage collected"
+    mkdir -p target/kibana-coverage/functional
+    mv target/kibana-coverage/server/coverage-final.json "$serverAndClientSummary"
+  fi
+
+  # Each browser unload event, creates a new coverage file.
+  # So, we merge them here.
+  if [[ -d "$KIBANA_DIR/target/kibana-coverage/functional" ]]; then
+    echo "--- Merging code coverage for FTR Config: $config"
+    yarn nyc report --nycrc-path src/dev/code_coverage/nyc_config/nyc.functional.config.js --reporter json
+    rm -rf target/kibana-coverage/functional/*
+    mv target/kibana-coverage/functional-combined/coverage-final.json "$functionalSummary"
+  else
+    echo "--- Code coverage not found"
+  fi
+
+  # Check for empty summary files.
+  empties=()
+  emptyCheck() {
+    echo "### Checking $1"
+    echo $(head -5 $1) | grep -E -i "pct.+Unknown" >/dev/null
+    lastCode=$?
+    if [ $lastCode -eq 0 ]; then
+      echo "  --- Empty Summary File: $1"
+      empties+=($1)
+    fi
+  }
+  emptyCheck $serverAndClientSummary
+  emptyCheck $functionalSummary
+
+  if [[ ${#empties[@]} -ge 2 ]]; then
+    echo "### Empty count = ${#empties[@]}, fail the build"
```

Comment:
@dmlemeshko also, we are going to have to allow "soft failures" on ingestion now too.
Because, @LeeDr wants to be able to have partial ingestions. 

I guess we could simply not fail in the ingest step, if it cant find a summary file.